app/javascript/dashboard/components/widgets/conversation/contextMenu/Index.vue

```diff
@@ -0,0 +1,264 @@
+<template>
+  <div class="conversation-menu">
+    <!-- // Status Menu  -->
+    <template v-for="item in menuItems">
+      <div
+        v-if="show(item.key)"
+        :key="item.key"
+        class="menu-item flex-align-center"
+        @click="toggleStatus(item.key, null)"
+      >
+        <fluent-icon :icon="item.icon" size="14" class="icon" />
+        <p>{{ item.label }}</p>
+      </div>
+    </template>
+    <!-- // Status Menu  -->
+    <!-- // Snooze Menu  -->
+    <div class="menu-item has-submenu">
+      <div class="flex-align-center">
+        <fluent-icon icon="snooze" size="14" class="icon" />
+        <p>{{ $t('CONVERSATION.CARD_CONTEXT_MENU.SNOOZE.TITLE') }}</p>
+      </div>
+      <fluent-icon icon="chevron-right" size="12" />
+      <div class="conversation-submenu">
+        <div class="menu-item" @click="toggleStatus(STATUS_TYPE.SNOOZED, null)">
```

Comment:
Make this a a new component (context-menu-with-submenu). Use the menu-item created previously for submenu.